054-infinite-auto-restart-of-Kafka-connectors.md

```diff
@@ -0,0 +1,160 @@
+# Infinite auto-restart of Apache Kafka connectors
+
+This proposal introduces changes to the connector auto-restart functionality which was introduced in [Proposal #7](https://github.com/strimzi/proposals/blob/main/007-restarting-kafka-connect-connectors-and-tasks.md) and implemented in [Strimzi 0.33.0](https://github.com/strimzi/strimzi-kafka-operator/pull/7500).
+
+## Current situation
+
+Strimzi 0.33.0 introduced support for automatically restarting connectors in Apache Kafka Connect clusters (when managed using the `KafkaConnector` resources). This feature is disabled by default, but can be enabled in the `KafkaConnectors` `.spec` section:
+
+```yaml
+apiVersion: strimzi.io/v1beta2
+kind: KafkaConnector
+metadata:
+  name: my-source-connector
+spec:
+  # ...
+  autoRestart:
+    enabled: true
+  # ...
+```
+
+When enabled, it works like this:
+* The connector is restarted up to 7 times
+* The times of the auto-restarts are:
+    * Immediate
+    * 2+ minutes after previous restart
+    * 6+ minutes after previous restart
+    * 12+ minutes …
+    * 20+ minutes …
+    * 30+ minutes …
+    * 42+ minutes …
+    * _There is no exact time, so it will be restarted when the next reconciliation happens and the time condition is fulfilled (i.e. restart after 6+ minutes can happen for example after 7 minutes in the next suitable periodical reconciliation)._
+* After 7th restart, the connector will not be automatically restarted anymore
+* When the container runs successfully for at least the backoff interval for the restart counter to be reset.
```

Comment:
Maybe mention what the backoff counter is?

Reply:
What do you mean by _backoff counter_ here?

Reply:
The _restart counter_?